src/AppBundle/Membership/AdherentChangePasswordHandler.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace AppBundle\Membership;
+
+use AppBundle\Entity\Adherent;
+use Doctrine\Common\Persistence\ObjectManager;
+use Symfony\Component\Security\Core\Encoder\EncoderFactoryInterface;
+
+class AdherentChangePasswordHandler
+{
+    private $manager;
+    private $encoderFactory;
+
+    public function __construct(ObjectManager $manager, EncoderFactoryInterface $encoderFactory)
```

Comment:
The service you mention just calls the factory, so no need for this level of indirection IMHO (see https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/SecurityBundle/Resources/config/security.xml#L53).